src/Features/Core/Portable/StackTraceExplorer/ParsedFrameWithFile.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    internal sealed class ParsedFrameWithFile : ParsedStackFrame
+    {
+        public TextSpan FileSpan { get; }
+
+        public ParsedFrameWithFile(
+            string originalLine,
+            TextSpan classSpan,
+            TextSpan methodSpan,
+            TextSpan argsSpan,
+            TextSpan fileSpan)
+            : base(originalLine, classSpan, methodSpan, argsSpan)
+        {
+            Contract.Requires(fileSpan.Length > 0);
+            FileSpan = fileSpan;
+        }
+
+        public string GetFileText()
```

Comment:
it's very weird to see a non virtual public method on the derived type.  indicates that the caller needs to understand which specific subclass this is to delegate to it.

Reply:
Removed this class and added to `ParsedStackFrame`